### PR TITLE
W-15346916-hpa-rtf-kt

### PR DESCRIPTION
**Pull request by kevintroller**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 1]
```diff
@@ -0,0 +1,241 @@
+= Configuring Horizontal Pod Autoscaling (HPA) for Runtime Fabric Deployments
+
+Runtime Fabric instances support horizontal scaling of Mule app deployments by spinning up additional replicas or spinning down unneeded replicas in response to the CPU usage as a signal.
+
+Mulesoft supports configuring Mule apps to be responsive to the CPU usage on Kubernetes (K8s) runtime side. You can enable CPU based horizontal autoscaling per application on the Runtime Manager application deployment page, which will result in automatic scale up or scale down of the deployment replicas as needed.
+
+In Kubernetes, a Horizontal Pod Autoscaler (HPA) automatically updates a workload resource, with the aim of automatically scaling the workload to match demand. Horizontal scaling means that the response to increased load is to deploy more pods.
+
+[NOTE]
+HPA is supported only on Runtime Fabric agent version 2.6.<>. and higher.
```
Comment (rithishapadmanabh) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]: @SparshAgarwal - what is the version that must go here?

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+* Scale up and scale down performance can vary based on the replica size and the application type.
+
+* The policy is optimized for replica sizes with <850m. Larger replica sizes might take longer to scale.
+
+=== Limitations
+
+There are some limitations to consider:
+
+* Horizontal autoscaling does not work with clustering and rate limiting.
```
Comment (rithishapadmanabh) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]: @kevintroller - this limitation has been removed now, can you please confirm with @SparshAgarwal

**Review by dmerlob — DISMISSED** [review]

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+----
+
+Therefore, the cluster autoscaler starts with:
+
+----
+- --balancing-ignore-label=k8s.amazonaws.com/eniConfigCustom
+- --balancing-ignore-label=vpc.amazonaws.com/eniConfig
+----
+
+=== Configure Overprovisioning Pod
```
Comment (TomoDC) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]:
```suggestion
=== Configure Overprovisioning Node
```
I think this should be over provisioning nodes, not pods.

Also need to apply the corresponding change in the text.

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+  namespace: app-namespace
+spec:
+  behavior:
+    scaleDown:
+      policies:
+      - periodSeconds: 15
+        type: Percent
+        value: 100
+      selectPolicy: Max
+      stabilizationWindowSeconds: 300
```
Comment (TomoDC) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]:
```suggestion
      stabilizationWindowSeconds: 1800
```
This is 1800.

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+      stabilizationWindowSeconds: 0
+  maxReplicas: 3
+  metrics:
+  - resource:
+      name: cpu
+      target:
+        averageUtilization: 70
+        type: Utilization
+    type: Resource
+  minReplicas: 1
```
Comment (TomoDC) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]:
```suggestion
  maxReplicas: <maxReplicasChosenFromUI>
  metrics:
  - resource:
      name: cpu
      target:
        averageUtilization: 70
        type: Utilization
    type: Resource
  minReplicas: <maxReplicasChosenFromUI>
```
min and max replicas are values that the customer can choose from UI, but the text before says that this is choosen by Mule and it is a hard value. We should change it so customers can understand that their values is not 1 and 3 (that 1 and 3 are examples, the same as the app name, but for example what it is under behaviour -> scaleUp and scaleDown, these are all fixed and owned by Mule).

Used this <maxReplicasChosenFromUI> but not sure how to reflect that minReplicas and maxReplicas are being set by the customer.

File: modules/ROOT/pages/configure-horizontal-autoscaling.adoc [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+    apiVersion: apps/v1
+    kind: Deployment
+    name: my-app
+----
+
+Some points to consider:
+
+Scale up can occur at most every 180 seconds. Each period, up to 100% of the currently running replicas may be added until the maximum configured replicas is reached. For scaling up there is no stabilization window. When the metrics indicate that the target should be scaled up, the target is scaled up immediately.
+
+Scale down can occur at most every 15 seconds. Each Period, up to 100% of the currently running replicas may be removed which means the scaling target can be scaled down to the minimum allowed replicas. The number of replicas removed is based on the aggregated calculations over the past 300 seconds of the stabilization window.
```
Comment (TomoDC) [inline modules/ROOT/pages/configure-horizontal-autoscaling.adoc]:
```suggestion
Scale down can occur at most every 15 seconds. Each Period, up to 100% of the currently running replicas may be removed which means the scaling target can be scaled down to the minimum allowed replicas. The number of replicas removed is based on the aggregated calculations over the past 1800 seconds of the stabilization window.
```

**Review by Cristian-Venticinque — APPROVED** [review]

**Pull request merged**